### PR TITLE
Font collection pagination: add min height to avoid infinite number

### LINKED ISSUE #59235: Font Library Pagination Footer JavaScript error: Uncaught RangeError: Invalid array length
### Description

I've been trying to find the best way to reproduce this issue for the past two days, and it doesn't always show up. I have not been able to pin point what exact conditions triggers the error.

When I open the Font Library and click on the tab panel "Install fonts", I see "The editor has encountered an unexpected error".
It happens both with and without fonts installed.

Here is the content from the "Copy Error" button:

```
RangeError: Invalid array length
    at PaginationFooter (http://localhost:8889/wp-includes/js/dist/edit-site.js?ver=9f909713a1d0e8a21a08:7452:20)
    at renderWithHooks (http://localhost:8889/wp-includes/js/dist/vendor/react-dom.js?ver=18.2.0:16315:20)
    at updateFunctionComponent (http://localhost:8889/wp-includes/js/dist/vendor/react-dom.js?ver=18.2.0:19598:22)
    at beginWork (http://localhost:8889/wp-includes/js/dist/vendor/react-dom.js?ver=18.2.0:21611:18)
    at beginWork$1 (http://localhost:8889/wp-includes/js/dist/vendor/react-dom.js?ver=18.2.0:27436:16)
    at performUnitOfWork (http://localhost:8889/wp-includes/js/dist/vendor/react-dom.js?ver=18.2.0:26567:14)
    at workLoopSync (http://localhost:8889/wp-includes/js/dist/vendor/react-dom.js?ver=18.2.0:26476:7)
    at renderRootSync (http://localhost:8889/wp-includes/js/dist/vendor/react-dom.js?ver=18.2.0:26444:9)
    at recoverFromConcurrentError (http://localhost:8889/wp-includes/js/dist/vendor/react-dom.js?ver=18.2.0:25860:22)
    at performSyncWorkOnRoot (http://localhost:8889/wp-includes/js/dist/vendor/react-dom.js?ver=18.2.0:26106:22)
```

And here is the log.
[localhost-1708514725259.log](https://github.com/WordPress/gutenberg/files/14358714/localhost-1708514725259.log)

### Step-by-step reproduction instructions

In the Site Editor, open the Font Library from the Styles > Typopgraphy panel and click on the tab panel "Install fonts"

### Screenshots, screen recording, code snippet

_No response_

### Environment info

6.5-beta2-57673  on Local , using the beta tester nightly
6.5-beta2-57671-src which is my current copy of wordpress-develop.
With and without Gutenberg trunk.
Tested on Twenty Twenty-Four and on a custom block theme that does not add any font families.

-Untested on 6.4.3.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

No

### PR DESCRIPTION
**Pull request by matiasbenedetto**
## What?
 Add min height to the modal
 Fixes: https://github.com/WordPress/gutenberg/issues/59235

## Why?
to avoid infinite number

## How?
adding a constant with the min height.

## Testing Instructions
see: https://github.com/WordPress/gutenberg/issues/59235

## Screenshots or screencast 

https://github.com/WordPress/gutenberg/assets/1310626/761168e2-1e5c-4c32-b9cb-271dbbd538cc

### CODE REVIEW

File: packages/edit-site/src/components/global-styles/font-library-modal/font-collection.js [hunk 1]
```diff
@@ -118,7 +121,11 @@ function FontCollection( { slug } ) {
 
 	// NOTE: The height of the font library modal unavailable to use for rendering font family items is roughly 417px
 	// The height of each font family item is 61px.
-	const pageSize = Math.floor( ( window.innerHeight - 417 ) / 61 );
+	const windowHeight =
+		window.innerHeight <= MIN_WINDOW_HEIGHT
+			? MIN_WINDOW_HEIGHT
+			: window.innerHeight;
```
Comment (t-hamano) [inline packages/edit-site/src/components/global-styles/font-library-modal/font-collection.js]:
```suggestion
	const windowHeight = Math.max( window.innerHeight, MIN_WINDOW_HEIGHT );
```

I think this is probably a matter of preference, so I think either is fine.

**Review by t-hamano — APPROVED** [review]
LGTM!

**Comment by t-hamano** [comment]
@matiasbenedetto, If you rebase this PR, it should pass performance tests. See #59259

**Comment by getdave** [comment]
I just _manually_ cherry-picked this PR to the [cherry-pick-wp-6-5-beta-3 branch](https://github.com/WordPress/gutenberg/pull/59394) to get it included in the next release.

**Pull request merged**